### PR TITLE
XWIKI-19689: Support contains queries in SimpleEventQuery

### PR DESCRIPTION
**Pull request by michitux**
* Add a contains query condition
* Add the contains method in SimpleEventQuery
* Handle the contains query condition in the Solr event store
* Add tests

Jira issue: https://jira.xwiki.org/browse/XWIKI-19689

### CODE REVIEW

**Review by tmortagne — APPROVED** [review]

File: xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/query/CompareQueryCondition.java [hunk 1]
```diff
@@ -78,7 +78,14 @@ public class CompareQueryCondition extends AbstractPropertyQueryCondition
...
          */
-        ENDS_WITH
+        ENDS_WITH,
+
+        /**
+         * The property value contains the passed value.
+         *
+         * @since 14.4RC1
+         */
+        CONTAINS
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/query/CompareQueryCondition.java:88]: Shouldn't it be marked as unstable?

File: xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/query/SimpleEventQuery.java [hunk 2]
```diff
@@ -405,6 +405,19 @@ public SimpleEventQuery endsWith(String property, Object value)
         return this;
     }
 
+    /**
+     * @param property the name of the property
+     * @param value the value the property should contain
+     * @return this {@link SimpleEventQuery}
+     * @since 14.4RC1
+     */
+    public SimpleEventQuery contains(String property, Object value)
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/query/SimpleEventQuery.java:414]: Shouldn't it be marked as unstable?

File: xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/query/CompareQueryCondition.java (hunk 1 above)
Comment (michitux) [reply xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/query/CompareQueryCondition.java:88]: Good point, now I'm wondering why the other two recently added constants aren't marked as unstable.

Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/query/CompareQueryCondition.java:88]: > I'm wondering why the other two recently added constants aren't marked as unstable

I will go for the ravages of aging :)

**Pull request merged**